### PR TITLE
Add filter option to country choice type

### PR DESCRIPTION
**Pull request by stefandoorn**
| Q               | A
| --------------- | ---
| Bug fix?        | no|
| New feature?    | yes |
| BC breaks?      | no|
| License         | MIT |

### CODE REVIEW

**Comment by stefandoorn** [comment]
Hm, the logic existed previously. It might be deleted in the meanwhile - as I don't see it anymore now, but I had still some in my own project so that's what I referred to. In that case it doesn't make a lot of sense to merge this one for this sole location.

File: src/Sylius/Bundle/AddressingBundle/Form/Type/CountryChoiceType.php [hunk 1]
```diff
@@ -64,6 +68,7 @@ public function configureOptions(OptionsResolver $resolver): void
                 'enabled' => true,
                 'label' => 'sylius.form.address.country',
                 'placeholder' => 'sylius.form.country.select',
+                'filter' => null,
```
Comment (pamil) [inline src/Sylius/Bundle/AddressingBundle/Form/Type/CountryChoiceType.php]: Let's name it `choice_filter` to make it more self-explaining and consistent with `ZoneCodeChoiceType`.

**Comment by stefandoorn** [comment]
Alright. Will push a change tonight.

**Comment by stefandoorn** [comment]
P.S. @pamil Is that in the form of a Behat test wished or unit test?

**Comment by stefandoorn** [comment]
Adjusted code, not sure what is best approach to test, cannot find an example to follow for this kind of filter :(

**Comment by pamil** [comment]
@stefandoorn a unit test, you can take a look at `FixedCollectionTypeTest` or any `*TypeTest` in Sylius

**Comment by stefandoorn** [comment]
I tried to get it working @pamil using an example, with registering the form in 'getExtensions' etc., but I'm probably missing one thing to make it work. Error stays that the form gets 0 arguments injected in the form constructor. If you have a moment to insert an easy test, I can finish it afterwards.

File: src/Sylius/Bundle/AddressingBundle/phpunit.xml.dist [hunk 2]
```diff
@@ -13,6 +13,7 @@
     <testsuites>
         <testsuite name="SyliusAddressingBundle Test Suite">
             <directory>./test/</directory>
+            <directory>./Tests/</directory>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AddressingBundle/phpunit.xml.dist]: We should put these tests in a one folder I suppose

File: src/Sylius/Bundle/AddressingBundle/Tests/Form/Type/CountryChoiceTypeTest.php [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        $this->assertChoicesLabels(['France', 'Poland'], ['enabled' => null]);
+    }
+
+    /**
+     * @test
+     */
+    public function it_returns_countries_in_an_alphabetical_order()
+    {
+        $this->countryRepository->findBy(['enabled' => true])->willReturn([
+            $this->france->reveal(),
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AddressingBundle/Tests/Form/Type/CountryChoiceTypeTest.php]: We should probably change an order of countries to have a proper assert here

**Comment by stefandoorn** [comment]
Pushed adjustments based on feedback @lchrusciel.

File: src/Sylius/Bundle/AddressingBundle/phpunit.xml.dist (hunk 2 above)
Comment (pamil) [reply src/Sylius/Bundle/AddressingBundle/phpunit.xml.dist]: Not really, we use the following convention in other bundles (as Symfony does) and in fact `test` should be somehow merged to `Tests` (and merging it back to `test` breaks autoloading).

**Comment by stefandoorn** [comment]
Now it should be good again ;-)

File: src/Sylius/Bundle/AddressingBundle/phpunit.xml.dist (hunk 2 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/AddressingBundle/phpunit.xml.dist]: It is one of the solutions. I don't say, that we have to merge the new one with the existing one, but on the other hand we have a bundle test there. Just a thing, that we should discuss before we introduce too much test conventions in our test suite.

**Comment by pjedrzejewski** [comment]
Thank you Stefan!

**Pull request merged**